test/remote/push_db_tests.js

```diff
@@ -147,6 +149,7 @@ describe('remote push db', function() {
             assert.equal(device.pushCallback, '')
             assert.equal(device.pushPublicKey, '', 'device.pushPublicKey is correct')
             assert.equal(device.pushAuthKey, '', 'device.pushAuthKey is correct')
+            assert.equal(device.pushEndpointExpired, true, 'device.pushEndpointExpired is correct')
```

Comment:
It's probably worth adding an explicit test that updating with a new push endpoint, will reset this flag back to `false`.